### PR TITLE
894 select funding space

### LINKED ISSUE #894: Funding interface selects from available funding spaces
**Prerequisites:** #858, #876, #890 

**User story:** As a provider, I want to enroll kids in funding spaces as described in my contract. If I only have one kind of funding space for an age group, I shouldn't need to select it.

**AC:**

* Replace the dropdown menu with an interface as shown in the designs (component implemented by #890).
* Initially, the "Private pay" option is selected. Choosing "Child Day Care" reveals options to select a Contract space (showing the available FundingSpaces for the organization and age group) and a First funding period.
* If I only have one FundingSpace for the organization and age group, the dropdown menu is replaced by text indicating the type of space.
* Validations of the First funding period are the same as existing behavior.
* Validations require a FundingSpace is selected
* Validations prevent a user from associating a funding with a fundingspace that:
    * is not owned by the same organization that owns the funding enrollment site (fundingspace.organizationId == funding.enrollment.site.organizationId)
    * does not have the same agegroup as the funding enrollment (fundingspace.ageGroup == funding.enrollment.ageGroup)

## When there are multiple FundingSpaces

Screen Shot 2020-03-19 at 10 21 42 AM

## When there is only one FundingSpace for the organization and age group

Screen Shot 2020-03-19 at 10 21 46 AM

## Expanded dropdown with examples of formatting

Screen Shot 2020-03-19 at 10 21 50 AM

### PR DESCRIPTION
**Pull request by jlhogan**
closes #894 
has no new tests

### CODE REVIEW

File: src/Hedwig/ClientApp/src/components/ChoiceList/RadioButton.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default function RadioButton({
 	className,
 }: RadioButtonProps) {
 	return (
-		<div className={`usa-radio ${className}`}>
+		<div className={`usa-radio ${className ? className : ''}`}>
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/components/ChoiceList/RadioButton.tsx:23]: Let's us the `classnames` package

File: src/Hedwig/ClientApp/src/utils/models/fundingSpace.ts [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
 import { DeepNonUndefineable } from '../types';
+import { prettyFundingTime } from './fundingTime';
 
+export function prettyFundingSpaceTimeAllocations(fundingSpace: DeepNonUndefineable<FundingSpace>) {
+	if (!fundingSpace.fundingTimeAllocations) return '';
+
+	let str = '';
+	fundingSpace.fundingTimeAllocations.forEach((timeAllocation, idx) => {
+		if (idx > 0) str += ' / ';
+		str += `${prettyFundingTime(timeAllocation.time, idx == 0)} (${timeAllocation.weeks} weeks)`;
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/utils/models/fundingSpace.ts:11]: `===` strict equality

File: src/Hedwig/Repositories/SiteRepository.cs [hunk 3]
```diff
@@ -65,11 +65,18 @@ public Task<Site> GetSiteForOrganizationAsync(int id, int organizationId, string
 
 			return site.FirstOrDefaultAsync();
 		}
+
+		public Site GetSiteByIdAsNoTracking(int id)
```
Comment (cailyncodes) [inline src/Hedwig/Repositories/SiteRepository.cs:69]: Do we need the `asNoTracking` indicator?

Comment (jlhogan) [reply src/Hedwig/Repositories/SiteRepository.cs:69]: nope, left over from pre-always-no-tracking times!

**Pull request merged**